skyportal/handlers/api/source.py

```diff
@@ -934,3 +939,106 @@ async def get(self, obj_id):
 
                 # pause the coroutine so other handlers can run
                 await tornado.gen.sleep(1e-9)  # 1 ns
+
+
+class SourceAlertHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Send out a new source alert
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  additionalNotes:
+                    type: string
+                    description: |
+                      Notes to append to the message sent out
+                  groupIds:
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of IDs of groups whose members should get the alert (if they've opted in)
+                  sourceId:
+                    type: string
+                    description: |
+                      The ID of the Source's Obj the alert is being sent about
+                  level:
+                    type: string
+                    description: |
+                      Either 'soft' or 'hard', determines whether to send an email or email+SMS alert
+                required:
+                  - groupIds
+                  - sourceId
+                  - level
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+        """
+        if not cfg["alerts.enabled"]:
+            return self.error("Alerts are not enabled in current deployment.")
+        data = self.get_json()
+
+        if isinstance(data.get("additionalNotes"), str):
+            additional_notes = data["additionalNotes"].strip()
```

Comment:
```suggestion
            additional_notes = data["additionalNotes"].strip()
        else:
            additional_notes = None
```
Sorry, my initial suggestion was incomplete. Alternatively, can do all in one line with something like `addl_notes = data["additionalNotes"].strip() if isinstance(data.get("additionalNotes"), str) else None`